sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -716,7 +720,55 @@ abstract class FileParquetPartitionReaderBase(
         newColumns.safeClose()
       }
     } else {
-      inTable
+      inputTable
+    }
+  }
+
+  private def convertDecimal64ToDecimal32(cv: ColumnView, precisionList: Queue[Int]): ColumnView = {
```

Comment:
The problem is the List API in our Java bindings returns ColumnView so either I copy it out to a ColumnVector at every stage in case of a multi-nested list or just take a ColumnView as an input and do the dirty checks outside.